src/presenter/main-presenter.js

```diff
@@ -0,0 +1,150 @@
+    }
+
+    if (filtersFormView instanceof FiltersForm) {
+      this.#filtersFormView = filtersFormView;
+    } else {
+      throw new Error(`IllegalArgumentException! expected: ${FiltersForm}`);
+    }
+
+    if (sortFormView instanceof SortForm) {
+      this.#sortFormView = sortFormView;
+    } else {
+      throw new Error(`IllegalArgumentException! expected: ${SortForm}`);
+    }
+
+    if (noPointsMessageView instanceof NoPointsMessage) {
+      this.#noPointsMessageView = noPointsMessageView;
+    } else {
+      throw new Error(`IllegalArgumentException! expected: ${NoPointsMessage}`);
+    }
+
+    this.#controlsContainer = controlsContainer;
+    this.#eventsContainer = eventsContainer;
+  }
+
+  get points() {
+    if (this.#model === null) {
+      return [];
+    }
+    return this.#model.points;
+  }
+
+  get offers() {
+    if (this.#model === null) {
+      return [];
+    }
+    return this.#model.offers;
+  }
+
+  get destinations() {
+    if (this.#model === null) {
+      return [];
+    }
+    return this.#model.destinations;
+  }
+
+  init = () => {
+    render(this.#filtersFormView, this.#controlsContainer);
+    render(this.#sortFormView, this.#eventsContainer);
+  };
+
+  closeCurrentEditView = (pointPresenter) => {
+    if (this.#currentpointPresenter === null) {
+      this.#currentpointPresenter = pointPresenter;
+    } else {
+      replace(
+        this.#currentpointPresenter.pointRowView,
+        this.#currentpointPresenter.pointFormView
+      );
+      this.#currentpointPresenter = pointPresenter;
+    }
+  };
+
+  resetCurrentPointPresenter = () => {
+    this.#currentpointPresenter = null;
+  };
+
+  updatePointRowView = (pointPresenter) => {
+    const pointRowView = new PointRow(
+      pointPresenter.point,
+      this.#model.offers,
+      this.#model.destinations
+    );
+    replace(pointRowView, pointPresenter.pointRowView);
+    pointPresenter.pointRowView = pointRowView;
+  };
+
+  addPointToPointList = (point, pointRowView, pointFormView) => {
+    let pointPresenter;
+    if (this.#pointPresenters.has(point.id)) {
+      pointPresenter = this.#pointPresenters.get(point.id);
+    } else {
+      pointPresenter = new PointPresenter({
+        point,
+        pointRowView,
+        pointFormView,
+        pointsListView: this.#pointsListView,
+      });
+      this.#pointPresenters.set(point.id, pointPresenter);
+    }
+    pointPresenter.init(this.updatePointRowView, this.closeCurrentEditView, this.resetCurrentPointPresenter);
+    pointPresenter.renderPoint();
+  };
+
+  clearPointsList = () => {
+    remove(this.#pointsListView);
+  };
+
+  renderPointsList = (callback) => {
+    if (this.#model === null) {
+      return;
+    }
+    if (this.#model.points.length === 0) {
+      render(this.#noPointsMessageView, this.#eventsContainer);
+    } else {
+      callback.apply();
+      render(this.#pointsListView, this.#eventsContainer);
+    }
```

Comment:
а если сделать метод `#selectView`

```js
  #hasPoints = () =>this.#model.points.length === 0;
  #selectView = ()=> this.#hasPoints()?this.#noPointsMessageView:this.#pointsListView;
```

```suggestion
    //а тут останется
    render(this.#selectView(),this.#eventsContainer);
```

но вопрос. что за посторонний эффект callback? что-то пахнет керосином...

Reply:
по Бобу Мартину, пахнет плохим кодом ))